src/node_messaging.cc

```diff
@@ -1226,12 +1230,18 @@ JSTransferable::JSTransferable(Environment* env,
                                Local<Object> obj,
                                Local<Object> target)
     : BaseObject(env, obj) {
-  MakeWeak();
   target_.Reset(env->isolate(), target);
-  target_.SetWeak();
+}
+
+JSTransferable::~JSTransferable() {
+  HandleScope scope(env()->isolate());
```

Comment:
For education purposes: What is the purpose of initializing a scope in here?

Reply:
A local handle is created with `target_.Get(env()->isolate())` in the next line. A handle scope is needed in place to collect the local handle when it goes out of lexical scope.